tests/Fractions.Tests/StringFormatters/DecimalNotationFormatter_Format.cs

```diff
@@ -0,0 +1,822 @@
+    private static readonly string[] GeneralFormats =
+        ["G", "G0", "G1", "G2", "G3", "G4", "G5", "G6", "g", "g0", "g1", "g2", "g3", "g4", "g5", "g6"];
+
+    public static IEnumerable<TestCaseData> FormatWithTerminatingFractionCases =>
+        from stringFormat in GeneralFormats
+        from valueToTest in TerminatingFractions
+        from cultureToTest in FormatProviders
+        select new TestCaseData(valueToTest, stringFormat, cultureToTest)
+            .Returns(valueToTest.ToDouble().ToString(stringFormat, cultureToTest));
+
+    [Test]
+    [TestCaseSource(nameof(FormatWithTerminatingFractionCases))]
+    public string The_general_format_should_match_double_ToString_if_the_fraction_is_terminating(Fraction valueToTest, string stringFormat, IFormatProvider formatProvider) {
+        return DecimalFormatter.Format(stringFormat, valueToTest, formatProvider);
+    }
+
+    public static IEnumerable<TestCaseData> FormatWithNonTerminatingFractionAndSpecificPrecisionCases =>
+        from stringFormat in GeneralFormats.Except(["G", "g", "G0", "g0"])
+        from valueToTest in NonTerminatingFractions
+        from cultureToTest in FormatProviders
+        select new TestCaseData(valueToTest, stringFormat, cultureToTest)
+            .Returns(valueToTest.ToDouble().ToString(stringFormat, cultureToTest));
+
+    [Test]
+    [TestCaseSource(nameof(FormatWithNonTerminatingFractionAndSpecificPrecisionCases))]
+    public string The_general_format_should_match_double_ToString_when_the_precision_is_specified(Fraction valueToTest, string stringFormat, IFormatProvider formatProvider) {
+        return DecimalFormatter.Format(stringFormat, valueToTest, formatProvider);
+    }
+
+#if NET
+    public static IEnumerable<TestCaseData> FormatWithNonTerminatingFractionAndDefaultPrecisionCases =>
+        from stringFormat in new[] { "G", "g", "G0", "g0" }
+        from valueToTest in NonTerminatingFractions
+        from cultureToTest in FormatProviders
+        select new TestCaseData(valueToTest, stringFormat, cultureToTest);
+
+    [Test]
+    [TestCaseSource(nameof(FormatWithNonTerminatingFractionAndDefaultPrecisionCases))]
+    public void The_general_format_should_match_double_ToString_when_the_precision_is_not_specified(Fraction valueToTest, string stringFormat, IFormatProvider formatProvider) {
+        // Arrange
+        var doubleToString = valueToTest.ToDouble().ToString(stringFormat, formatProvider);
+
+        // Act
+        var fractionToString = DecimalFormatter.Format(stringFormat, valueToTest, formatProvider);
+
+        // Assert:
+        // while the fraction is correctly rounded to MidpointRounding.ToEven,
+        // there is apparently some optimization done to the double.ToString("g") that give weird results on the last decimal (cannot find any reference)
+        // (7.0/3).ToString(CultureInfo.InvariantCulture) returns "2,3333333333333335"
```

Comment:
This is due to the representation of 7/3 when using floating point arithmetic according to IEEE 754. Since we do not store the number in base 10, it is stored in an (unfortunately finite sequence, (1 + 2^(-1) + 2^( -2) .. + 2^(-52)) * 2^(exponent-K))

The rounding method inside `double` works correctly:

![image](https://github.com/danm-de/Fractions/assets/3825484/d75cbbc9-7b47-45c3-ac1b-5feca4df2d4c)

Reply:
The puzzling thing here was not related to our code, but rather the behavior of ToString with non-terminating decimals like (7.0/3) between .NET 48 and .NET8 - in the first case the value is always printed with the correct rounding, where as .NET8 appears to have "optimized" something and i cannot find when or what exactly (when using a precision specifier both frameworks act the same way) 😄 

> That's pretty cool. I wish this article from Microsoft had existed years ago (or it hadn't been so well hidden)